Storage

AmanEfer opened the pull request:
Create functionality fom interactive with products and stocks

AmanEfer committed:
Storage-1: Save point before creating further logic
- refactor TelegramBot class (create TelegramCommands class for it, deleted BotConfig class) and fix its test
- moved register user logic to StartCommand class (deleted RegisterCommand class and its test)
- prepared method for save stock in RestToCrud class, created StockDto class
- made some minor changes in other classes

AmanEfer committed:
Storage-1: Save point 2
- created StatesCache class
- made some minor changes in other classes

AmanEfer committed:
Storage-1: Create functionality for 'stock'
- created commands for save product, get all products, move products and also for create stock and get all stocks
- created classes with different caches
- create classes with user state, buttons and a couple of transfer data classes
- made some different changes in other classes
- temporarily commented out the tests in the telegram module cause going to change them in next commit

AmanEfer committed:
Merge branch 'develop' of https://github.com/AmanEfer/PP_3_1_4_Boot_Security_Bootstrap into storage

AmanEfer committed:
Storage-1: Create functionality for 'stock'
- added methods in ProductMapper from entity to dto and vice verca and mapper to Page
- added transient boolean field 'archived' and method 'isArchived()' in Product and Stock classes
- added method which finds products by name(not required) and returns value as Page
- made some minor changes in other classes

AmanEfer committed:
Storage-1: Create functionality for 'stock'
- created tests for commands classes
- made some minor changes in other classes

vicria commented on crud/src/main/java/com/amanefer/crud/controllers/ProductController.java:
```diff
@@ -66,6 +69,16 @@ public List<ProductDto> getAllProducts() {
+                                                 @RequestParam(required = false, defaultValue = "0") int pageNumber,
+                                                 @RequestParam(required = false, defaultValue = "3") int pageSize) {
+
+        return title == null ?
```
я б предложила добавить еще одну перегрузку и это условие проверить в сервисе

AmanEfer replied:
Исправил

vicria commented on crud/src/main/java/com/amanefer/crud/entities/Stock.java:
```diff
@@ -30,9 +31,12 @@ public class Stock {
     private LocalDateTime deletedAt;
 
+    @Transient
+    private boolean archived;
```
когда такие поля есть у нескольких сущностей, создается абстрактный родитель, куда выносятся все те поля, которые дублируются. или интерфейсы - архивируемый, датируемый и подобное

AmanEfer replied:
Исправил

vicria commented on crud/src/main/java/com/amanefer/crud/mappers/UserMapper.java:
```diff
@@ -23,4 +23,10 @@ public interface UserMapper extends BaseMapper<UserDto, UserModel, User> {
     UserDto fromModelToDto(UserModel userModel);
 
+    @Override
+    UserDto fromEntityToDto(User user);
```
может создать интерфейс для мапперов?

AmanEfer replied:
Интерфейс был. Удалил все переопределенные методы

vicria commented on crud/src/main/java/com/amanefer/crud/repositories/ProductRepository.java:
```diff
@@ -12,6 +15,12 @@ public interface ProductRepository extends JpaRepository<Product, String> {
 
     List<Product> findAllByDeletedAtIsNull();
 
+    @Query("from Product p where p.deletedAt is null")
```
без Query не получится такое сделать?

AmanEfer replied:
Исправил

vicria commented on telegram/pom.xml:
```diff
@@ -45,5 +45,17 @@
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.httpcomponents.client5</groupId>
```
а это зачем?

AmanEfer replied:
Без этого не работает patch запрос в rest template. По-другому у меня не получается никак сделать, выдает ошибку, что patch не используется в rest template

vicria commented on telegram/src/main/java/com/amanefer/telegram/cache/ProductStateCache.java:
```diff
@@ -0,0 +1,26 @@
+@Component
+public class ProductStateCache implements BotCache<Long, ProductDto> {
+
+    Map<Long, ProductDto> cache = new HashMap<>();
```
а почему не использовать настоящий кеш?)

AmanEfer replied:
Это я не умею еще. Если потом дойду до этого, то буду пробовать исправлять

vicria commented on telegram/src/main/java/com/amanefer/telegram/commands/product/ProductsCommand.java:
```diff
@@ -0,0 +1,79 @@
+                .callbackData(Button.GET_ALL_PRODUCTS_BUTTON.toString())
+                .build();
+
+        InlineKeyboardButton moveProductsButton = InlineKeyboardButton.builder()
```
а нельзя эти три кнопки сделать финальными статическими? зачем их каждый раз создавать?

AmanEfer replied:
Исправил

AmanEfer committed:
Storage-1: Create functionality for 'stock'
- enabled flyway in crud pom

AmanEfer committed:
Storage-1: Create functionality for 'stock'
- created base entity for Product and Stock
- deleted override methods from mappers
- added empty list check for stocks
- changed getPageOfAllProducts() in ProductServiceImpl
- rename project
- made some minor changes

AmanEfer committed:
Storage-1: Create functionality for 'stock'
- added attribute 'mainClass' in all poms
- created dockerfile for filesExporter module and added it ib docker-compose
- refactored some command classes (made buttons static final)
- added check for empty list for products
- made some minor changes
